pms/models/pms_reservation.py

```diff
@@ -1213,38 +1215,29 @@ def _compute_shared_folio(self):
     @api.depends("partner_id", "partner_id.name")
     def _compute_partner_name(self):
         for record in self:
-            if record.partner_id and not record.partner_name:
-                record.partner_name = record.partner_id.name
-            elif not record.partner_name:
-                record.partner_name = False
+            self.folio_id._apply_partner_name(record)
```

Comment:
this is a compute, the folio does not have to exist even when the field is calculated, so you should call the method using the model through api.model: self.env ["pms.folio"] ._ apply_x (record )